pyproject.toml

```diff
@@ -33,18 +33,18 @@ dependencies = [
     "numpy>=1.24",
     "pandas>=2.0",
     "xarray>=2023.04",
-    "netCDF4",
-    "packaging",
+    "netCDF4>=1.7",
+    "packaging>=22.0",
 ]
 dynamic = ["version"]
 
 [project.optional-dependencies]
 all = [
-    "contextily",
-    "geopandas",
-    "IPython",  # 'ipython' is not the correct module name.
-    "pyarrow",
-    "rioxarray",
+    "contextily>=1.2",
+    "geopandas>=0.14",
+    "IPython>=8",  # 'ipython' is not the correct module name.
+    "pyarrow>=16",
```

Comment:
Pinning to `pyarrow>16` here to fix this error:

```python-traceback
________ test_to_numpy_pyarrow_array_pyarrow_dtypes_string[string_view] ________
>   ???
E   KeyError: 'string_view'
pyarrow/types.pxi:5025: KeyError
During handling of the above exception, another exception occurred:
dtype = 'string_view'
    @pytest.mark.skipif(not _HAS_PYARROW, reason="pyarrow is not installed")
    @pytest.mark.parametrize(
        "dtype",
        [
            None,
            "string",
            "utf8",  # alias for string
            "large_string",
            "large_utf8",  # alias for large_string
            "string_view",
        ],
    )
    def test_to_numpy_pyarrow_array_pyarrow_dtypes_string(dtype):
        """
        Test the _to_numpy function with PyArrow arrays of PyArrow string types.
        """
>       array = pa.array(["abc", "defg", "12345"], type=dtype)
../pygmt/tests/test_clib_to_numpy.py:333: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pyarrow/array.pxi:230: in pyarrow.lib.array
    ???
pyarrow/types.pxi:5040: in pyarrow.lib.ensure_type
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
>   ???
E   ValueError: No type alias for string_view
pyarrow/types.pxi:5027: ValueError
```

Code was added in https://github.com/GenericMappingTools/pygmt/pull/3608:

https://github.com/GenericMappingTools/pygmt/blob/97185e8437fe73cc8661e89a3f655ea2b51f25f4/pygmt/tests/test_clib_to_numpy.py#L326-L336

The [`pyarrow.StringViewArray`](https://arrow.apache.org/docs/16.0/python/generated/pyarrow.StringViewArray.html) class was added in https://github.com/apache/arrow/pull/39652 that was released in [pyarrow v16.0](https://arrow.apache.org/blog/2024/04/20/16.0.0-release/)

Reply:
PyArrow v16.0 was released in April, 2024. Instead of pinning pyarrow>=16.0, we can just skip `string_view` if pyarrow<v16.0.

Reply:
Ok, I just tested older versions from pyarrow v12 to v15, and think we can pin to `pyarrow>=13` ([v13 was released on 24 Aug 2023](https://arrow.apache.org/blog/2023/08/24/13.0.0-release/))  if ignoring `string_view`. With v12, there is another error on datetime.

```python-traceback
_________________________________________________________________ test_to_numpy_pyarrow_array_pyarrow_dtypes_date[date64[ms]] __________________________________________________________________

dtype = 'date64[ms]', expected_dtype = 'datetime64[ms]'

    @pytest.mark.skipif(not _HAS_PYARROW, reason="pyarrow is not installed")
    @pytest.mark.parametrize(
        ("dtype", "expected_dtype"),
        [
            pytest.param("date32[day]", "datetime64[D]", id="date32[day]"),
            pytest.param("date64[ms]", "datetime64[ms]", id="date64[ms]"),
        ],
    )
    def test_to_numpy_pyarrow_array_pyarrow_dtypes_date(dtype, expected_dtype):
        """
        Test the _to_numpy function with PyArrow arrays of PyArrow date types.

        date32[day] and date64[ms] are stored as 32-bit and 64-bit integers, respectively,
        representing the number of days and milliseconds since the UNIX epoch (1970-01-01).

        Here we explicitly check the dtype and date unit of the result.
        """
        data = [
            date(2024, 1, 1),
            datetime(2024, 1, 2),
            datetime(2024, 1, 3),
        ]
        array = pa.array(data, type=dtype)
        result = _to_numpy(array)
        _check_result(result, np.datetime64)
>       assert result.dtype == expected_dtype  # Explicitly check the date unit.
E       AssertionError: assert dtype('<M8[D]') == 'datetime64[ms]'
E        +  where dtype('<M8[D]') = array(['2024-01-01', '2024-01-02', '2024-01-03'], dtype='datetime64[D]').dtype

../pygmt/tests/test_clib_to_numpy.py:364: AssertionError
```

The main change appears to be in https://github.com/apache/arrow/issues/33321, when pyarrow supported preserving the datetime64 temporal resolution.

Maybe we can keep the `string_view` test, but add a `skip_if(pyarrow.__version__ < 16)` marker or something like that.